File: 2025_12/leetcode2025-12-20.cpp
#include <iostream>
#include <vector>
#include <string>
using namespace std;

class Solution {
public:
    int minDeletionSize(vector<string>& strs) {
        int m = strs.size(), n = strs[0].size();
        int res = 0;
        for (int j=0; j<n; j++){
            for (int i=0; i<m-1; i++){
                if (strs[i][j] > strs[i+1][j]){
                    res++;
                    break;
                }
            }
        }
        return res;
    }
};

int main(){
    Solution s;
    vector<string> strs = {"cba","daf","ghi"};
    cout << s.minDeletionSize(strs) << endl;
}

File: 2025_12/leetcode2025-12-20.py
from typing import List

class Solution:
    def minDeletionSize(self, strs: List[str]) -> int:
        m, n = len(strs), len(strs[0])
        res = 0
        for j in range(n):
            for i in range(m-1):
                if strs[i][j] > strs[i+1][j]:
                    res += 1
                    break
        return res

if __name__ == "__main__":
    s = Solution()
    strs = ["cba","daf","ghi"]
    print(s.minDeletionSize(strs))
